src/Strategy.sol

```diff
@@ -46,48 +108,138 @@ contract Strategy is BaseStrategy {
+            uint256 _unstakedLPTokens = unstakedLPBalance(); // How many are available to unstake?
+            uint256 _requiredLPTokensToUnstake; // How many more LP tokens do we need to unstake to get the required amount of `want`.
+
+            // Free up the minimum amount of LP tokens to get to the amount of `want` we need
+            if (_unstakedLPTokens < _lpTokensToSell) {
+                _requiredLPTokensToUnstake =
+                    _lpTokensToSell -
+                    _unstakedLPTokens;
+                if (_stakedLpTokens >= _requiredLPTokensToUnstake) {
+                    _unstakeLPTokens(_requiredLPTokensToUnstake);
+                } else if (_stakedLpTokens > 0) {
+                    _unstakeLPTokens(_stakedLpTokens);
+                }
+            }
+
+            //withdraw from pool
+            uint256 _unstakedLpBalance = unstakedLPBalance();
+            if (_unstakedLpBalance > 0) {
+                _withdrawLiquidity(
+                    unstakedLPBalance(),
+                    _minAmountOfLPToWant(_unstakedLpBalance)
+                );
             }
+
+            _liquidWant = wantBalance();
+        }
+
+        if (_amountNeeded > _liquidWant) {
+            _liquidatedAmount = _liquidWant;
+            _loss = _amountNeeded - _liquidWant;
         } else {
             _liquidatedAmount = _amountNeeded;
         }
     }
 
     function liquidateAllPositions() internal override returns (uint256) {
-        // TODO: Liquidate all positions and return the amount freed.
-        return want.balanceOf(address(this));
+        // unstake all staked token
+        synStakingMC.emergencyWithdraw(pid, address(this));
+        uint256 _lpTokenBalance = unstakedLPBalance();
+        if (_lpTokenBalance > 0) {
+            // Try to withdraw everything in `want`
+            _withdrawLiquidity(_lpTokenBalance, 0);
+
+            _lpTokenBalance = unstakedLPBalance();
+            // If we still have LP tokens after trying to withdraw everything in `want`,
+            // withdraw in anything else that we can get
+            if (_lpTokenBalance > 0) {
+                uint256[] memory minAmounts = new uint256[](3);
+                syn3PoolSwap.removeLiquidity(
+                    _lpTokenBalance,
+                    minAmounts,
+                    block.timestamp
+                );
+            }
+        }
+
+        return wantBalance();
     }
 
     // NOTE: Can override `tendTrigger` and `harvestTrigger` if necessary
-    // solhint-disable-next-line no-empty-blocks
+
     function prepareMigration(address _newStrategy) internal override {
-        // TODO: Transfer any non-`want` tokens to the new strategy
+        nUSD.transfer(_newStrategy, nusdBalance());
+        USDT.transfer(_newStrategy, usdtBalance());
+
+        // Calim all our SYN tokens and transfer
+        synStakingMC.harvest(pid, address(this));
```

Comment:
Best practice:
remove any non-critical logic like this from `prepareMigration`. 
reason is that if this function is borked (chances of borking increase when we have more external contract calls / dependencies), then we have no way to recover.

Suggestion here could be to remove this line and just to harvest prior to migration.